kadi/commands/commands_v2.py

```diff
@@ -353,43 +359,61 @@ def update_archive_and_get_cmds_recent(
             cmd_event["Date"], cmd_event["Event"], cmd_event["Params"]
         )
 
-        # Events that do not generate commands (e.g. load interrupt) return
-        # None from get_cmds_from_event.
+        # Events that do not generate commands return None from
+        # get_cmds_from_event.
         if cmds is not None and len(cmds) > 0:
             cmds_list.append(cmds)
-            rltts.append(None)
 
     # Sort cmds_list and rltts by the date of the first cmd in each cmds Table
     cmds_starts = np.array([cmds["date"][0] for cmds in cmds_list])
     idx_sort = np.argsort(cmds_starts)
     cmds_list = [cmds_list[ii] for ii in idx_sort]
-    rltts = [rltts[ii] for ii in idx_sort]
 
-    for ii, cmds, rltt in zip(itertools.count(), cmds_list, rltts):
-        if rltt is not None:
-            # Apply RLTT from this load to the current running loads (cmds_all).
-            # Remove commands with date greater than the RLTT date. In most
+    # Apply RLTT and any END SCS commands (CODISAXS) to loads. RLTT is applied
+    # by stopping SCS's 128-133.
+    for ii, cmds in enumerate(cmds_list):
+        cmds_source = cmds["source"][0]
+        if cmds_source == "CMD_EVT":
+            cmds_source = f"CMD_EVT {cmds[0]['event']} at {cmds[0]['date']}"
+        logger.info(f"Processing {cmds_source} with {len(cmds)} commands")
+        end_scs = collections.defaultdict(list)
+        if date_end := cmds.meta.get("rltt"):
+            source = f'RLTT in {cmds["source"][0]}'
+            end_scs[date_end, source].extend([128, 129, 130, 131, 132, 133])
+
+        # Explicit END SCS commands. Most commonly these come from command events
+        # like SCS-107, but can also be in weekly loads e.g. disabling an ACIS
+        # ECS measurement in SCS-135.
+        ok = cmds["tlmsid"] == "CODISASX"
+        if np.any(ok):
+            for cmd in cmds[ok]:
+                if (scs := cmd["params"]["codisas1"]) >= 128:
```

Comment:
Why are these limited to >= 128 here?  Because that's where they start or none below that would be relevant?

Reply:
None below 128 would be relevant. The SCS slots below 128 are protected SCS's that are in the OBC protected memory and require an OBC patch to change. The commands archive does not deal with starting and stopping those.

Reply:
OK, so is this in here to be more self-documenting?  Because there also aren't any of those < 128 commands in here to be filtered, right?

Reply:
There are SCS 98 disables in the weekly loads.

Reply:
Right, but no scs=98 commands _in_ the loads.

Reply:
Correct.